src/local.rs

```diff
@@ -447,4 +469,52 @@ mod tests {
             panic!("unexpected error type: {:?}", err);
         }
     }
+
+    #[tokio::test]
+    async fn invalid_path() {
+        let root = TempDir::new().unwrap();
+        let root = root.path().join("🙀");
+
+        // Invalid paths supported above root of store
+        let integration = LocalFileSystem::new(root.clone());
+
+        let directory = Path::from("directory");
+        let object = directory.child("child.txt");
+        integration
+            .put(&object, Bytes::from("arbitrary"))
+            .await
+            .unwrap();
+        integration.head(&object).await.unwrap();
+        integration.get(&object).await.unwrap();
+
+        flatten_list_stream(&integration, None).await.unwrap();
+        flatten_list_stream(&integration, Some(&directory))
+            .await
+            .unwrap();
+
+        let result = integration.list_with_delimiter(&directory).await.unwrap();
+        assert_eq!(result.objects.len(), 1);
+        assert!(result.common_prefixes.is_empty());
+        assert_eq!(result.objects[0].location, object);
+
+        let illegal = root.join("💀");
+        std::fs::write(illegal, "foo").unwrap();
+
+        // Can list directory that doesn't contain illegal path
+        flatten_list_stream(&integration, Some(&directory))
+            .await
+            .unwrap();
+
+        // Cannot list illegal file
+        let err = flatten_list_stream(&integration, None)
```

Comment:
I initially thought this should be:

```rust
        let err = flatten_list_stream(&integration, Some(&illegal))
```

Edit: However, then I realized `flatten_list_stream` is really listing all entries starting at `prefix` which (now) includes the illegal filename